pyro/poutine/reparam_messenger.py

```diff
@@ -80,10 +102,10 @@ def _pyro_sample(self, msg):
         # similarly be safely applied twice, with the second application
         # avoiding overwriting the original application.
         for m in _get_init_messengers():
```

Comment:
E.g. here it is not expected that `_get_init_messengers()` can return `None`. See the comment below about the return type of the `effectful`.